### PR TITLE
update italy spider

### LINKED ISSUE #822: Italy: Parse JSON instead of HTML
The page requests this document: https://www.appaltipop.it/_next/data/LxpUO4Pg-S_nnq33fzaED/it/tenders.json

### PR DESCRIPTION
**Pull request by Ravf95**
Closes #822

### CODE REVIEW

File: kingfisher_scrapy/spiders/italy.py [hunk 1]
```diff
@@ -18,14 +18,26 @@ class Italy(SimpleSpider):
 
     def start_requests(self):
         yield scrapy.Request(
-            'https://www.appaltipop.it/it/download',
-            meta={'file_name': 'list.html'},
+            'https://www.appaltipop.it/_next/data/LxpUO4Pg-S_nnq33fzaED/it/tenders.json',
```
Comment (yolile) [inline kingfisher_scrapy/spiders/italy.py:21]:
```suggestion
            # From https://www.appaltipop.it/it/download
            'https://www.appaltipop.it/_next/data/LxpUO4Pg-S_nnq33fzaED/it/tenders.json',
```

File: kingfisher_scrapy/spiders/italy.py [hunk 2]
```diff
@@ -18,14 +18,26 @@ class Italy(SimpleSpider):
...
+        #   "pageProps": {
+        #       other fields,
+        #       "buyers": [ ... ],
+        #       other fields
+        #    },
+        #   "__N_SSG": ...
+        # }
+
+        for buyer in data['pageProps']['buyers']:
+            resource = buyer['appaltipop:releases/0/buyer/dataSource/resources'][0]
```
Comment (yolile) [inline kingfisher_scrapy/spiders/italy.py:40]:
```suggestion
            # The first resource in the list is the OCDS JSON, the second one a XLSX file
            resource = buyer['appaltipop:releases/0/buyer/dataSource/resources'][0]
```

File: kingfisher_scrapy/spiders/italy.py [hunk 3]
```diff
@@ -18,14 +18,26 @@ class Italy(SimpleSpider):
...
+            filepath = resource['appaltipop:releases/0/buyer/resource/url']
```
Comment (yolile) [inline kingfisher_scrapy/spiders/italy.py:41]:
```suggestion
            # The JSON file path looks like 'data/IT-CF-01232710374/ocds.json'
            filepath = resource['appaltipop:releases/0/buyer/resource/url']
```

File: kingfisher_scrapy/spiders/italy.py (hunk 1 above)
Comment (Ravf95) [reply kingfisher_scrapy/spiders/italy.py:21]: Done

File: kingfisher_scrapy/spiders/italy.py (hunk 2 above)
Comment (Ravf95) [reply kingfisher_scrapy/spiders/italy.py:40]: Done

File: kingfisher_scrapy/spiders/italy.py (hunk 3 above)
Comment (Ravf95) [reply kingfisher_scrapy/spiders/italy.py:41]: Done

File: kingfisher_scrapy/spiders/italy.py [hunk 4]
```diff
@@ -18,14 +18,30 @@ class Italy(SimpleSpider):
...
+        #    },
+        #   "__N_SSG": ...
+        # }
+
+        for buyer in data['pageProps']['buyers']:
+            # The first resource in the list is the OCDS JSON, the second one a XLSX file
+            resource = buyer['appaltipop:releases/0/buyer/dataSource/resources'][0]
+
+            # The JSON file path looks like 'data/IT-CF-01232710374/ocds.json'
+            filepath = resource['appaltipop:releases/0/buyer/resource/url']
```
Comment (yolile) [inline kingfisher_scrapy/spiders/italy.py:45]:
```suggestion
            file_path = resource['appaltipop:releases/0/buyer/resource/url']
```

File: kingfisher_scrapy/spiders/italy.py [hunk 5]
```diff
@@ -18,14 +18,30 @@ class Italy(SimpleSpider):
...
+            url = f'https://raw.githubusercontent.com/ondata/appaltipop/master/{filepath}'
```
Comment (yolile) [inline kingfisher_scrapy/spiders/italy.py:46]:
```suggestion
            url = f'https://raw.githubusercontent.com/ondata/appaltipop/master/{file_path}'
```

**Review by yolile — APPROVED** [review]

**Pull request merged**